pipeline-adapters/mlrun-pipelines-kfp-common/src/mlrun_pipelines/common/ops.py

```diff
@@ -680,3 +685,141 @@ def _enrich_node_selector(function):
+    )
+    process.start()
+    result = queue.get()
+    process.join()
+    return result
+
+
+def enrich_kfp_workflow_credentials(
+    byte_buffer: bytes,
+    content_type: str,
+    env_vars: list[V1EnvVar],
+) -> bytes:
+    if content_type.endswith("zip"):
+        modified_zip_bytes = _enrich_kfp_workflow_zip_credentials(
+            yaml_bytes=byte_buffer,
+            env_vars=env_vars,
+        )
+        return modified_zip_bytes
+    elif content_type.endswith(("yaml", "plain")):
+        modified_yaml_bytes = _enrich_kfp_workflow_yaml_credentials(
+            yaml_bytes=byte_buffer,
+            env_vars=env_vars,
+        )
+        return modified_yaml_bytes
+    else:
+        raise ValueError(f"Unsupported content type {content_type}")
+
+
+def _enrich_kfp_workflow_zip_credentials(
+    byte_buffer: bytes,
+    env_vars: list[V1EnvVar],
+) -> bytes:
+    in_memory_zip = io.BytesIO(byte_buffer)
+    with zipfile.ZipFile(in_memory_zip, "r") as zip_read:
+        file_list = zip_read.namelist()
+        files_data = {}
+        for file_name in file_list:
+            with zip_read.open(file_name) as f:
+                files_data[file_name] = f.read()
+
+    for file_name, file_data in files_data.items():
+        if file_name.endswith(("yaml", "plain")):
+            modified_yaml = _enrich_kfp_workflow_yaml_credentials(
+                yaml_bytes=file_data,
+                env_vars=env_vars,
+            )
+            files_data[file_name] = modified_yaml
+
+    out_memory_zip = io.BytesIO()
+    with zipfile.ZipFile(out_memory_zip, "w") as zip_write:
+        for file_name, file_data in files_data.items():
+            zip_write.writestr(file_name, file_data)
+
+    return out_memory_zip.getvalue()
+
+
+def _enrich_kfp_workflow_yaml_credentials(
+    yaml_bytes: bytes,
+    env_vars: list[V1EnvVar],
+) -> bytes:
+    """
+    Modifies the given workflow dictionary to add environment variables from a Kubernetes secret
+    to the container specifications.
+    The function checks if the workflow is compatible with KFP v1 or KFP v2 and adds the
+    environment variables accordingly.
+
+    Parameters:
+    - workflow_dict: dict representing the workflow
+
+    Returns:
+    - None (the dict is modified in place)
+    """
+    workflow_dict = yaml.safe_load(yaml_bytes)
+    # Determine the KFP version by checking the 'apiVersion' field
+    api_version = (
+        workflow_dict.get("api_version") or workflow_dict.get("apiVersion", "").lower()
+    )
+
+    if "argoproj.io" in api_version:  # KFP Argo Workflow
+        for template in workflow_dict["spec"]["templates"]:
+            if "container" in template:
+                container = template["container"]
+                for env_var in env_vars:
+                    env_var = create_env_for_container(env_var)
+                    if "env" in container and container["env"] is not None:
+                        container["env"].append(env_var)
```

Comment:
I think it would actually be better to check if the env already contains a variable of that name, and if it does - replace it. Appending blindly may cause a situation where you have multiple copies of the same env var, which can cause unpredictable results.